### PR TITLE
if request is needed for the cache, force get request on redirect uri…

### PR DESCRIPTION
**Pull request by chrift**
… - ensures the correct cookies will be available to the request

Patch bump
Test updates

### CODE REVIEW

File: lib/routes.js [hunk 1]
```diff
@@ -39,17 +40,23 @@ module.exports = (
...
     path: config.redirectUri,
     config: {
       auth: false
     },
     handler: async (request, h) => {
       try {
-        const { payload } = request
+        const { payload, method, query } = request
+
+        if (method === 'post' && config.passRequestToCacheMethods) {
```
Comment (brendanarnold) [inline lib/routes.js]: Can you explain in comments what you did in idm-tech? i.e. why it redirects to self?

File: lib/routes.js [hunk 2]
```diff
@@ -39,17 +40,23 @@ module.exports = (
...
+          return h.redirect(`${config.redirectUri}?${qs.stringify(payload)}`)
+        }
+
+        const requestParams = method === 'get' ? query : payload
 
         /** Get our saved state **/
-        const { state } = payload
+        const { state } = requestParams
```
Comment (brendanarnold) [inline lib/routes.js]: Will the app have an unhandled crash if you do a GET without a query with 'state' in it - could happen if robot crawled site

Comment (chrift) [reply lib/routes.js]: Query is always available as an empty object to get requests

File: lib/routes.js (hunk 1 above)
Comment (brendanarnold) [inline lib/routes.js]: Is method always lowercase?

Comment (chrift) [reply lib/routes.js]: Yes

**Pull request merged**